### PR TITLE
Gulp ruby integration

### PR DESCRIPTION
**Pull request by chalettu**
This commit is for issue
https://github.com/ManageIQ/manageiq-ui-service/issues/259 which asked
to be able to automatically launch a miq backend when gulp serve is
run.

### CODE REVIEW

File: gulp/tasks/manageiq-server.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = function (gulp, options) {
+    var manageiq_dir = require('../config')[options.key || 'manageiq_dir'];
+
+    return task;
+
+    function task() {
+
+          var cmd = spawn('rails', ['s'], {stdio: 'inherit',cwd:manageiq_dir});
+    }   
+};
```
Comment (martinpovolny) [inline gulp/tasks/manageiq-server.js]: missing endline, please, fix you editor setting.

**Comment by himdel** [comment]
Please don't. SSUI should never try to start the manageiq app. It should depend on the app already running.

Most devels already work with manageiq started, sometimes on diffferent ports, this is not something SSUI should ever do.

Also, so many style checks must have failed on that file.. was there any need to rush this so much?

**Comment by martinpovolny** [comment]
Honestly I have to say that I don't like the speed of merge here.

**Comment by martinpovolny** [comment]
By the way what about hose Merge commits here? I am quite sure we have previously refused to merge PR with such issues.

**Comment by chriskacerguis** [comment]
@chalettu @AllenBW Martin brings up a good point here about style checks.  It looks like it passed Travis / CodeClimate, but there appear to be some style issues.  Can you please take a look at our eslint rules, travis job, etc to ensure that we have everything setup as needed?

@chalettu Can you please address the style comments in a new PR, and please run ESlint to make sure that all are good.

@himdel RE: the MIQ app already running...since gulp serve is just run in development, I'm not sure that it really matters who starts what.  That said, I can think of 1 use case where that could be an issue.  What are your concerns with gulp serve starting MIQ?  The SUI team met here in Austin, TX and we liked the idea of having MIQ started by gulp.

@himdel RE: Style checks.  Every test passed, so if something was missed we should look at that (see comment to @chalettu and @AllenBW)

@martinpovolny RE: merge commits.  Yep, my bad there.  I spoke to @Fryguy about that, and we decided to just leave it (as @Fryguy said "it happens").  I have had @chalettu update his settings to ensure that doesn't happen again.

@martinpovolny RE: speed of merge.  The team is here in Austin (with @Fryguy and @chessbyte), so we were all discussing things and so we were able to move a bit faster (but, see comment above, I did make a mistake by merging the "merges" in the PR.

**Comment by AllenBW** [comment]
@himdel @chriskacerguis code climate excludes checking the following folders
- tests/**/*
- node_modules/**
- gulp/**
- bower_components/**
- reports/**

furthermore  we only run eslint on app files, not build tasks

thinking we might want this changed ?

**Comment by chriskacerguis** [comment]
@AllenBW Yeah, since we are going to move to Node, then we should check the Node server side, and the gulp tasks.

**Comment by martinpovolny** [comment]
> Most devels already work with manageiq started, sometimes on diffferent ports,

I +1 that. I personally run different ManageIQ's on different ports from different directories so that I can comfortably switch versions and also switch between doing review/testing and working on my stuff.

Now I will have a thing that will try to run one of my copies of ManageIQ by default. I really don't like that.

I quite like to run SUI as the docs suggest:

```
PROXY_HOST=127.0.0.1:3000 gulp serve
```

that allows me to select the desired MIQ server by providing the port number.

Now I have the SUI reach out of it's directory and try to run MIQ in ../manageiq/ even if I provide port for a different MIQ server. Even one that is on a different host.

Maybe this fit's someones simple usecase, but for me this is counterintuitive. 

Best way to resolve all the issues with this PR would be reverting it. 

Minimal mitigation should include fixing the style problems and either: 
 * running the MIQ server only if the PROXY_HOST was not given or
 * provide a 2nd 'serve' task maybe 'run" that does not start the MIQ server.

**Pull request merged**